lib/express-rate-limit.js

```diff
@@ -27,52 +53,41 @@ function RateLimit(options) {
         }
     });
 
+    options = defaults(options, {
+      store: new MemoryStore(options.windowMs), // store to use for persisting rate limit data
+    });
+
 
     if (options.global) {
         throw new Error('The global option was removed from express-rate-limit v2.');
     }
 
 
-    // this is shared by all endpoints that use this instance
-    var hits = {};
-
-
     function rateLimit(req, res, next) {
         var key = options.keyGenerator(req, res);
 
-        if (hits[key]) {
-            hits[key]++;
-        } else {
-            hits[key] = 1;
-        }
-
-        if (options.max && hits[key] > options.max) {
-            return options.handler(req,res, next);
-        }
-
-        if (options.delayAfter && options.delayMs && hits[key] > options.delayAfter) {
-            var delay = (hits[key] - options.delayAfter) * options.delayMs;
-            setTimeout(next, delay);
-        } else {
-            next();
-        }
-    }
-
-    function resetAll() {
-        hits = {};
+        options.store.incr(key, function(err, current) {
+          if (err) {
+            return next(err);
+          }
+
+          if (options.max && current > options.max) {
+              return options.handler(req,res, next);
+          }
+
+          if (options.delayAfter && options.delayMs && current > options.delayAfter) {
+              var delay = (current - options.delayAfter) * options.delayMs;
+              setTimeout(next, delay);
+          } else {
+              next();
+          }
+        });
     }
 
-    // simply reset ALL hits every windowMs
-    setInterval(resetAll, options.windowMs);
-
-    // export an API to allow hits from one or all IPs to be reset
-    function resetKey(key) {
-        delete hits[key];
-    }
+    rateLimit.resetKey = options.store.resetKey;
```

Comment:
this should have it's scope bound:

    rateLimit.resetKey = options.store.resetKey.bind(options.store);

 (MemoryStore doesn't happen to use `this` but presumably other stores could.)